CleverRaven/Cataclysm-DDA
# Hells Raiders (roof) fixup

#### Summary
None



#### Purpose of change
Fix Hell's Raiders so their facilities get roofs without relying in roof magic.



#### Describe the solution
- Added roofs to the bandit camp, including the tents.
- Changed bandit camp floors from natural rock to a sensible choice.
- Fixed up the bandit drug lab, where the source drug lab was still left in the file (so this copy was removed), and the regular drug lab roof was added to the facility.
- Added roofs to the bandit labor camp, including the tents.
- Changed the labor camp stoves to sit on a floor rather than on a barbed wire fence (overriding palette).



#### Describe alternatives you've considered
Add vent pipes onto the labor camp stoves. Decided against it due to it being a shoddy piece of work.



#### Testing
- Load suitable start save.
- Debug spawn facility.
- Check if it's OK.
- Fix up the JSON if not and repeat until it is OK.
- Debug spawn facility... until all facilities have been examined and (optionally) fixed.
![Screenshot (439)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/d5fe301c-62d4-4ea3-8913-18516cf5f19a)
![Screenshot (440)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/e077489b-5530-4467-be6e-a8d15ed09b3e)
![Screenshot (441)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/b747dc17-6548-4db3-8398-54863ca7d2ce)
![Screenshot (442)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/ea1da977-bc46-48fe-bc1f-a173093e864b)
![Screenshot (443)](https://github.com/CleverRaven/Cataclysm-DDA/assets/22739822/857d577c-5e61-4d87-8208-256514ba7d5f)
The hawk eyed will note that the last two screen shots weren't from the same session, as the barbed wire/stove issue was noted (and fixed) after the roof screenshot was made, and I couldn't be bothered repeating it for a local change to a different map.



#### Additional context
Only the tents used by the affected maps were being given roofs, as those were the only ones being tested. The rest will be dealt with when getting to that file (and/or its usages). It can also be noted that these are prime candidates for conversion to usage of 3D functionality when available so you'd only place the tents on the ground level and the roofs would move with them if the placement is adjusted.
